### ISSUE TITLE
not portable, if root folder move to new location will return error: MicroMamba hook not found. 

### ISSUE THREAD

**Issue by bryanray** [oobabooga/text-generation-webui#852]
not portable, if root folder move to new location will return error: MicroMamba hook not found.

**Comment by jllllll** [comment]
Did you also move `installer_files`? That is where the virtual environment is located and is where the hook is.

**Comment by bryanray** [comment]
yes, i did, the entire folder is moved, to a usb device, so it can plug in to high performance gpu, but it out put this error, i am confused.

**Comment by jllllll** [comment]
This must be a bug in Micromamba. If the hook is not found, Micromamba is called to regenerate the hook. However, if you moved the entirety of `installer_files`, then the hook should still be there. Check to see if `\installer_files\mamba\condabin\activate.bat` exists and that `\installer_files\env` is not empty.

You may need to use the Miniconda version of the installer:
https://github.com/jllllll/one-click-installers/tree/oobabooga-windows-miniconda

Micromamba is still in development, so Miniconda may work better for this purpose.

**Comment by bryanray** [comment]
thanks

what i did was, i copy paste the eintire root folder, /oobabooga-windows, everything in it...
so it should not be me missing anything.

**Comment by bryanray** [comment]
it will need to be reinstalled, every time i want to move it some where else, and it is freaking annoying,
even I change the root folder name, it will still output this stupid ( echo. && echo Micromamba failed to download. && goto end )

**Comment by jllllll** [comment]
I saw some information online saying that installing miniconda the way that the previously mentioned installer does will create a portable environment. However, this may have been false or misleading. It may not be possible to create a conda environment that is truly portable due to parts of conda being coded to use absolute paths.

**Comment by jllllll** [comment]
Try deleting `\installer_files\mamba\condabin`. That is where the hook is, and that hook seems to contain absolute paths to the original install. Deleting that folder should cause the script to regenerate the hook with new paths in the new location.

**Comment by bryanray** [comment]
thanks a lot, i was trying to figure out where the hook was...
\installer_files\mamba\condabin. 
i will see if it work once i rename or change path...

why this project have to use the awkward conda or manba....why cant it be good old python.....

**Comment by jllllll** [comment]
It's not like you can't use pip to install everything. There are instructions for that on the front page. The installer uses micromamba because it doesn't require installation and is useful for managing packages. None of these things were ever meant to be portable.

**Comment by bryanray** [comment]
well.. webui from a1111 can be portable i can confirm. it has system folder as env and in side ins just python and git...

thanks again, 
mamba_hook.bat
micromamba.bat
does contain absolute path....
i am so annoyed by this non potable approach :))
